rest-services/client/src/test/java/com/_4point/aem/docservices/rest_services/client/generatedPDF/RestServicesGeneratePDFServiceAdapterTest.java

```diff
@@ -0,0 +1,290 @@
+									Integer.toString(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()))),
+					() -> assertThat(exMsg,
+							containsStringIgnoringCase(Response.Status.INTERNAL_SERVER_ERROR.getReasonPhrase())),
+					() -> assertThat(exMsg, containsStringIgnoringCase(HTML_ERROR_PAGE_HTML)));
+		}
+
+		private static void validateNoEntity(String exMsg) {
+			assertAll(() -> assertThat(exMsg, containsStringIgnoringCase("Call to server succeeded")),
+					() -> assertThat(exMsg, containsStringIgnoringCase("server failed to return createPDFResult xml")));
+		}
+
+		private static void validateHtmlResponse(String exMsg) {
+			assertAll(() -> assertThat(exMsg, containsStringIgnoringCase("Response from AEM server was")),
+					() -> assertThat(exMsg, containsStringIgnoringCase("content-type")),
+					() -> assertThat(exMsg, containsStringIgnoringCase(MediaType.TEXT_HTML)));
+		}
+
+		private static void validateNoContentTypeInResponse(String exMsg) {
+			assertAll(() -> assertThat(exMsg, containsStringIgnoringCase("Response from AEM server was")),
+					() -> assertThat(exMsg, containsStringIgnoringCase("content-type")),
+					() -> assertThat(exMsg, containsStringIgnoringCase("null")));
+		}
+	}
+
+	@ParameterizedTest
+	@EnumSource
+	void testCreatePDF(ErrorResponseScenario scenario) throws Exception {
+		scenario.restClientMocksSetup.accept(this);
+		GeneratePDFServiceBuilder generatePDFServiceBuilder = RestServicesGeneratePDFServiceAdapter.builder()
+				.machineName(TEST_MACHINE_NAME).port(TEST_MACHINE_PORT).basicAuthentication("username", "password")
+				.useSsl(false).clientFactory(() -> client);
+		underTest = generatePDFServiceBuilder.build();
+		byte[] inputBytes = "document Data".getBytes();
+		Document inputDoc = MockDocumentFactory.GLOBAL_INSTANCE.create(inputBytes);
+		Document settingsDoc = null;
+		Document xmpDoc = null;
+		GeneratePDFServiceException ex = assertThrows(GeneratePDFServiceException.class,
+				() -> underTest.createPDF2(inputDoc, "pptx", "Filetype Settings", PDFSettings.PDFA1b_2005_RGB,
+						SecuritySettings.Certificate_Security, settingsDoc, xmpDoc));
+		String msg = ex.getMessage();
+		assertNotNull(msg);
+		scenario.validateExceptionMessage.accept(msg);
+	}
+
+	@Test
+	void testConvertXmlToCreatePDFResult() throws Exception {
+		CreatePDFResult createPDFResult = RestServicesGeneratePDFServiceAdapter
+				.convertXmlToCreatePDFResult(new ByteArrayInputStream(POPULATED_CREATE_PDF_RESULT_XML.getBytes()));
+		validatePopulatedResult(createPDFResult);
+	}
+
+	@Test
+	void testConvertXmlToAssemblerEmptyResult() throws Exception {
+		CreatePDFResult result = RestServicesGeneratePDFServiceAdapter
+				.convertXmlToCreatePDFResult(new ByteArrayInputStream(EMPTY_CREATE_PDF_RESULT_XML.getBytes()));
+		validateEmptyResult(result);
+	}
+
+	private void validateEmptyResult(CreatePDFResult result) {
+		assertTrue(result.getCreatedDocument().isEmpty());
+		assertTrue(result.getLogDocument().isEmpty());
+	}
+
+}
```

Comment:
I notice in the test coverage that we're missing some coverage that would be pretty easy to add.  Can we add the following scenarios:
1) Null inputDoc  - should generate NullPointerException
2) Null inputFileExtensions - should generate NullPointerException

Reply:
Done